repo/content/index/info.go

```diff
@@ -21,7 +21,119 @@ type Info struct {
 	EncryptionKeyID     byte                 `json:"encryptionKeyID,omitempty"`
 }
 
+type InfoCompact struct {
+	PackBlobID          *blob.ID
```

Comment:
given that `blob.ID` is a `string` would it make sense to do interning instead of having a pointer here? IOW, can we avoid this structure altogether because it would be identical to the `Info`, we just need to be careful to share the string instead of having all these copies?

Reply:
golang string copy assignment is always deep copy.
So if we use `blob.ID`, we will not able to share the memory kept in `packBlobIDs` of `oneUseBuilder`.

Moreover, whenever we copy `Info`, a new string with the whole new memory will be allocated. Even though we always be careful to dereference the source of the copy, the memory footprint also increases considering the delay of GC.

Reply:
i'm pretty sure strings share the underlying data when assigning and even slicing.

To get a deep copy of strings, you need to use `strings.Clone()` https://pkg.go.dev/strings#Clone

Reply:
You are right. `StringData` is lazy copied only when the shared data is to be changed.
For the current case, `packBlobID` is never changed, so it is fine to use `string`.

And for the same reason, my previous change, which use *string for `packBlobID`, makes no help to the memory save, instead it increases memory footprint of string since it always allocs a new string for each `Info`.

Reply:
I changed the code accordingly and tested it again, now the peek memory usage reduces to 6GB:

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                              
1185499 root      20   0   10.7g   6.0g 860144 S 348.2  19.1  26:04.75 kopia